scripts/process_uploads.py

```diff
@@ -0,0 +1,161 @@
+    Only one .csv file exists in the upload bucket and previous batch directory.
+    """
+
+    client = storage.Client()
+    bucket = client.get_bucket(bucket_name)
+
+    all_blobs = list(client.list_blobs(bucket_name, prefix=prefix))
+    csv_path = list(filter(lambda blob: blob.name.endswith('.csv'), all_blobs))[0].name
+
+    blob = bucket.get_blob(csv_path).download_as_text()
+    csv_reader = csv.DictReader(io.StringIO(blob))
+
+    samples: List[str] = []
+    for row in csv_reader:
+        samples.append(row['sample.sample_name'])
+
+    return samples, csv_path
+
+
+def determine_samples(
+    upload_bucket: str, upload_prefix: str, main_bucket: str, previous_batch_prefix: str
+):
+    """Determine files that should be moved to main vs archive bucket.
+    Determines the difference between the latest CSV file within upload
+    and the CSV file from the most recent batch as the list of samples to
+    be moved.
+    """
+    all_samples: List[str] = []
+    previous_samples: List[str] = []
+
+    all_samples, csv_path_upload = samples_from_csv(upload_bucket, upload_prefix)
+    previous_samples, _ = samples_from_csv(main_bucket, previous_batch_prefix)
+
+    samples = set(all_samples) - set(previous_samples)
+
+    curr_csv_file_path = join(upload_bucket, csv_path_upload)
+
+    return samples, curr_csv_file_path
+
+
+def generate_file_list(samples: List[str]):
+    """ Generate list of expected files, given a list of sampleIDs """
+    main_files: List[str] = []
+    archive_files: List[str] = []
+    for s in samples:
+        main_files.extend([f'{s}.g.vcf.gz', f'{s}.g.vcf.gz.tbi', f'{s}.g.vcf.gz.md5'])
+        archive_files.extend([f'{s}.cram', f'{s}.cram.crai', f'{s}.cram.md5'])
+
+    return main_files, archive_files
+
+
+@click.command()
+@click.option('--batch_number')
+def run_processor(
+    batch_number: int,
+):
+    """Set up and execute batch_move_files
+
+    Parameters
+    =========
+    batch_number: int
+    Indexed starting at 0. Used to navigate subdirectory structure
+    e.g. 0 --> batch0
+
+    """
+
+    # Setting up inputs for batch_move_files
+    project = os.getenv('HAIL_BILLING_PROJECT')
+    batch_path = f'batch{batch_number}'
+    prev_batch = f'batch{int(batch_number)-1}'
+    upload_bucket = f'cpg-{project}-upload'
+    upload_prefix = ''
+    upload_path = join(upload_bucket, upload_prefix)
+    main_bucket = f'cpg-{project}-main'
+    main_prefix = join('gvcf', batch_path)
+    main_path = join(main_bucket, main_prefix)
+    prev_prefix = join('gvcf', prev_batch)
+    archive_path = join(f'cpg-{project}-archive', 'cram', batch_path)
+
+    docker_image = os.environ.get('DRIVER_IMAGE')
+    key = os.environ.get('GSA_KEY')
+
+    # Determine files to be processed
+    samples, csv_path = determine_samples(
+        upload_bucket, upload_prefix, main_bucket, prev_prefix
+    )
+    main_files, archive_files = generate_file_list(samples)
+
+    # Initialize the service backend.
+    hl.init()
+
```

Comment:
I think it should actually work without this.

```suggestion
    hl.init()

```